src/hotspot/share/gc/serial/markSweep.cpp

```diff
@@ -172,12 +172,12 @@ void MarkSweep::mark_object(oop obj) {
     _string_dedup_requests->add(obj);
   }
 
+  ContinuationGCSupport::transform_stack_chunk(obj);
```

Comment:
Add a comment, something like: "// Do the transform while we still have the header intact, which might include important class information".